### PR TITLE
refactor(typescript): popper, button & placeholder components

### PR DESCRIPTION
**Pull request by philibea**
## Type

- Refactor

### Summarize concisely:

Refactor Popper, Button, Placeholder

### CODE REVIEW

File: src/components/Placeholder/Block.tsx [hunk 1]
```diff
@@ -30,7 +30,7 @@ const styles = {
   `,
 }
 
-const Block = props => (
+const Block: React.VoidFunctionComponent = props => (
```
Comment (chambo-e) [inline src/components/Placeholder/Block.tsx:33]:
```suggestion
const Block: VoidFunctionComponent = props => (
```

File: src/components/Button/__tests__/__snapshots__/index.js.snap [hunk 2]
```diff
@@ -106,7 +106,6 @@ exports[`Button should render correctly loading 1`] = `
 <button
     aria-disabled="false"
     class="emotion-0 emotion-1 emotion-2"
-    type="button"
```
Comment (DorianMaliszewski) [inline src/components/Button/__tests__/__snapshots__/index.js.snap:109]: Why we dropped this ?

File: src/components/Button/__tests__/__snapshots__/index.js.snap [hunk 3]
```diff
@@ -1294,6 +1285,7 @@ exports[`Button should render correctly with an action button 1`] = `
   -webkit-flex-shrink: 0;
   -ms-flex-negative: 0;
   flex-shrink: 0;
+  border-radius: 0px;
```
Comment (DorianMaliszewski) [inline src/components/Button/__tests__/__snapshots__/index.js.snap:1288]: Is this wanted ?

File: src/components/Button/__tests__/__snapshots__/index.js.snap (hunk 2 above)
Comment (philibea) [reply src/components/Button/__tests__/__snapshots__/index.js.snap]: You right it's fixed and typed

File: src/components/Button/__tests__/__snapshots__/index.js.snap (hunk 3 above)
Comment (philibea) [reply src/components/Button/__tests__/__snapshots__/index.js.snap]: Same, it's fixed

File: .gitignore [hunk 4]
```diff
@@ -7,3 +7,4 @@ coverage
 storybook-static
 .reports
 yarn-error.log
+.vscode/*
```
Comment (chambo-e) [inline .gitignore:10]: 💩

File: src/components/Placeholder/BoxWithIcon.tsx [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+const boxStyled = (theme: Theme) => css`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+  border: 1px solid ${theme.colors.gray300};
+`
+
+const BoxWithIcon: FunctionComponent<{
```
Comment (chambo-e) [inline src/components/Placeholder/BoxWithIcon.tsx:23]:
```suggestion
const BoxWithIcon: VoidFunctionComponent<{
```

Comment (philibea) [reply src/components/Placeholder/BoxWithIcon.tsx:23]: Fixed @chambo-e

**Review by chambo-e — APPROVED** [review]

**Review by vincentaudebert — APPROVED** [review]

**Pull request merged**